### ISSUE TITLE
no-this-in-sfc false positives

### ISSUE THREAD

**Issue by milesj** [yannickcr/eslint-plugin-react#1967]
The following code triggers ESLint errors on the `this.reset()` line.

```
class Foo {
  unsubscribe(ticketID) { 
    return dispatch => {
      this.reset();
      dispatch(forceUpdate());

      return null;
    };
  }
}
```

There is no React or JSX in this file.

**Comment by ljharb** [comment]
Thanks, definitely a bug.

cc @alexzherdev

**Comment by alundiak** [comment]
+1

**Comment by alundiak** [comment]
@sergei-startsev , @ljharb is that change going to be released as `v7.11.2` ?

**Comment by ljharb** [comment]
It will be included in the next version, whatever that is and whenever it's ready.
